payu/models/cesm_cmeps.py

```diff
@@ -190,6 +184,73 @@ def setup(self):
                 # TODO: copied this from other models. Surely we want to exit here or something
                 print('payu: error: Unable to find mod_def.ww3 file in input directory')
 
+    def _setup_checks(self):
+        # check pelayout fits within requested cpucount
+        cpucount = int(
+            self.expt.config.get('ncpus', multiprocessing.cpu_count())
+        )
+        all_realms = self.realms
+        for realm in all_realms:
+            ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
+            nthreads = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_nthreads"))
```

Comment:
We should include this for future proofing

Reply:
I’m not sure. It is not currently in use and its removal would streamline the code.

If, in the future, Gadi provides cpus with hyperthreading or if we need nthreads, we can certainly revisit its inclusion then. For now, I believe removing it will help keep our codebase more focused on our immediate needs.

Or we could leave a comment, such as `#not used`, to indicate thread is not in use directly in `nuopc.runconfig`, which is similar to what we have done for other variables.

Reply:
It can't be 0, so its not really "not used".

I would definitely like to investigate threading in cice, and think the option might get used.

Reply:
I've tried -1 and 0, both work.

Reply:
Oh sorry , yes - but I guess that just means the model has overwritten the setting ?

Reply:
Yes, it is overwritten by `maxthreads`, which [defaults to be 1](https://github.com/ESCOMP/CMEPS/blob/5b7d76978e2fdc661ec2de4ba9834b985decadc6/cesm/driver/esm.F90#L930).

Reply:
I added an if nthread>1 case which captures this